tests_and_analysis/test/euphonic_test/test_brille_interpolator.py

```diff
@@ -0,0 +1,294 @@
+        grid = get_brille_grid(grid_file)
+        with pytest.raises(expected_exception):
+            BrilleInterpolator(faulty_crystal, grid)
+
+    # Deliberately create a grid incompatible with the quartz crystal
+    # and check that it raises an error
+    @pytest.mark.parametrize('expected_exception, grid_file',
+        [(ValueError, 'lzo_trellis_10.hdf5'),
+         (ValueError, 'quartz_mesh_10_unfilled.hdf5')])
+    def test_faulty_grid_object_creation(self, expected_exception, grid_file):
+        grid = get_brille_grid(grid_file)
+        with pytest.raises(expected_exception):
+            BrilleInterpolator(get_crystal('quartz'), grid)
+
+
+class TestBrilleInterpolatorCalculateQpointPhononModes:
+
+    @pytest.mark.parametrize(
+        'material, from_fc_kwargs, emax, expected_sf1d_file', [
+            ('LZO', {'grid_npts': 100}, 100,
+             'La2Zr2O7_trellis_100_sf_1d_average.json')
+        ])
+    def test_calculate_qpoint_phonon_modes(
+            self, material, from_fc_kwargs, emax, expected_sf1d_file):
+        fc = get_fc(material)
+        bri = BrilleInterpolator.from_force_constants(fc, **from_fc_kwargs)
+
+        qpts = get_test_qpts()
+        qpm = bri.calculate_qpoint_phonon_modes(qpts)
+        # Replace frequencies to ensure same binning
+        qpm_eu = fc.calculate_qpoint_phonon_modes(qpts)
+        qpm.frequencies = qpm_eu.frequencies
+
+        # Calculate structure factor 1D to test eigenvectors
+        sf = qpm.calculate_structure_factor()
+        ebins = np.arange(5, emax)*ureg('meV')
+        sf1d = sf.calculate_1d_average(ebins)
+        expected_sf1d = get_spectrum1d(expected_sf1d_file)
+        check_spectrum1d(sf1d, expected_sf1d, y_rtol=0.01, y_atol=3e-3)
+
+    def test_brille_qpoint_phonon_modes_similar_to_those_from_fc(self):
+        fc = get_fc('graphite')
+        bri = BrilleInterpolator.from_force_constants(fc, grid_npts=5000)
+        qpts = np.array([[-0.20, 0.55, 0.55],
+                         [ 0.35, 0.07, 0.02],
+                         [ 0.00, 0.50, 0.00],
+                         [ 0.65, 0.05, 0.25],
+                         [ 1.80, 0.55, 2.55]])
+        qpm_fc = fc.calculate_qpoint_phonon_modes(qpts)
+        qpm_brille = bri.calculate_qpoint_phonon_modes(qpts)
+        # Replace frequencies to ensure same binning
+        qpm_brille.frequencies = qpm_fc.frequencies
+        # Calculate structure factor 1D to test eigenvectors
+        ebins = np.arange(5, 190)*ureg('meV')
+        sf_fc = qpm_fc.calculate_structure_factor()
+        sf1d_fc = sf_fc.calculate_1d_average(ebins)
+        sf_brille = qpm_brille.calculate_structure_factor()
+        sf1d_brille = sf_brille.calculate_1d_average(ebins)
+        # Tolerances are quite generous, but that is required unless
+        # we have a very dense grid (expensive to test)
+        check_spectrum1d(sf1d_brille, sf1d_fc,
+                         y_rtol=0.01, y_atol=5e-3)
```

Comment:
I think a 1% tolerance is fine, actually, considering this is meant to be a fast approximate method. It seems to be better that what was obtained in the LZO example in the documentation where at low energies the relative frequency residual seems to be around 2% (from the graph, a residual of 0.15meV at around 7meV).

Reply:
Yeah, this test actually only creates a spectrum starting from 5 meV because I thought trying to reliably test those low frequencies would be a bit of a nightmare